tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/source_code/load_monaco_shim_test.ts

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {loadMonaco, windowWithRequireAndMonaco} from './load_monaco_shim';
+
+describe('loadMonaco shim', () => {
+  function createFakeRequire(): Require {
+    let require = ((modules: string[], callback: Function) => {
+      callback();
+    }) as any;
+    return Object.assign(require, {
+      config: () => {},
+      toUrl: () => {},
+      defined: () => {},
+      specified: () => {},
+      onError: () => {},
+      undef: () => {},
+      onResourceLoad: () => {},
+    });
+  }
+
+  // TODO(cais): Explore better typing by depending on external libraries.
+  function createFakeMonaco() {
+    return {};
+  }
+
+  let requireSpy: jasmine.Spy;
+  beforeEach(() => {
+    windowWithRequireAndMonaco.require = createFakeRequire();
```

Comment:
Use proper jasmine API? 
```ts
spyOnProperty(window, 'require').and.returnValue({
  theRequireMethodWithWeirdProperties
});
```

Reply:
I tried that and it didn't work, because `window.require` doesn't exist or have an accessor type.